# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1235458113 -3600
#      Tue Feb 24 07:48:33 2009 +0100
# Branch release-3-0-x
# Node ID a08f22ff5ba18419453e528fcfe006d795576750
# Parent  e896fba7e3d481614727529034ef88c8e88834b6
fplot.m: fix nargin check

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-02-23  John W. Eaton  <jwe@octave.org>
+
+	* plot/fplot.m: Fix nargin check.
+	From Joel Parker <Joel.Parker@radiancetech.com>.
+
 2008-01-15  Rafael Laboissiere  <rafael@debian.org>
 
 	* gethelp.cc: Include <cstdio>.
 
 2009-02-14  Tatsuro MATSUOKA <tmacchant@yahoo.co.jp>
 
 	* miscellaneous/fileparts.m: Fix again.
 
diff --git a/scripts/plot/fplot.m b/scripts/plot/fplot.m
--- a/scripts/plot/fplot.m
+++ b/scripts/plot/fplot.m
@@ -35,17 +35,17 @@
 ##    fplot ("[cos(x), sin(x)]", [0, 2*pi])
 ## @end example
 ## @seealso{plot}
 ## @end deftypefn
 
 ## Author: Paul Kienzle <pkienzle@users.sf.net>
 
 function fplot (fn, limits, n, linespec)
-  if (nargin < 2 || nargin > 3)
+  if (nargin < 2 || nargin > 4)
     print_usage ();
   endif
 
   if (nargin < 3) 
     n = 0.002; 
   endif
 
   have_linespec = true;
